### ISSUE TITLE
Byes

### ISSUE THREAD

**Issue by fbatista** [Juizes-MTG-Portugal/Juizes-MTG-Portugal.github.io#2]
Add changes to the MTR Addendum so that byes become fair.

Inspiration: [Evaluating_Byes_mindfreak314.pdf](https://github.com/Juizes-MTG-Portugal/Juizes-MTG-Portugal.github.io/files/13426131/Evaluating_Byes_mindfreak314.pdf)

**Comment by FranciscoBarata21** [comment]
after reading the article the 4 points for bye looks alright, but maybe its worth trying out 3 points for tournaments with 6+ rounds of swiss.  Not sure if that makes sense or not.

**Comment by fbatista** [comment]
Current state:

- Bye worth 5 points, but with a tiebreaker lower than actual wins.

Proposal:

Idea A: ❤️ 

- Bye worth 4 points.

---

Idea B: 😄 

- Bye worth a variable amount based on the number of rounds (and consequentially the number of players)

---

Idea C: 🎉 

- Bye worth a variable amount based on which round # it was awarded. Early byes worth less, later byes worth more.

---

Idea D: 🚀 

- Don't award byes at all, and instead use pods of 3 players, while ensuring that a player can only be paired in a pod of 3 at most once per tournament.

**Comment by mindfreak314** [comment]
Author of the initial article here. I'm currently running more simulations for bigger tournaments, since the behaviour is very dependant on the number of rounds and what cut is used. I'm currently sticking to the structure recommended by the [TopDeck Championship Series](https://docs.google.com/document/d/1BUei_Vi338ZwvdOHvWjK71MONLJ57r4cwywPi57ntV4/edit#heading=h.w2xiwh1j8igg), since this is the main software in use atm. This structure is not equivalent to the one in the MSTR for tournaments with 15-32 players and the top cuts also differ (as of now CT only supports Top4 and Top16). It's also woth noting, that not all TOs abide by these reccommendations in one or another way.
Will report back once simulations for all variations are done for Eminence and MSTR sturcture, in order to give feedback on what could be done right now within the possibilities of the CT software and what could be done outside of that. I also want to compare with empirical data. All of that will take some time though. For now, simply using 4 instead of 5 points should already be a big improvement regardless of structure.

**Comment by fbatista** [comment]
I'm also working on modifying the simulator i built back in the day, to make it something that we can run a high number of iterations and get statistical data from:

https://gist.github.com/fbatista/5d811ea1f28ea30a16ecdeb1c67d2c51

I appreciate any suggestion of data to analyse.

So far i got this going:

https://docs.google.com/spreadsheets/d/1QhiYzBfBhckK2AyAKM68sjJLUuYIBJlZYwnrZuzpsXI/edit?usp=sharing

**Comment by mindfreak314** [comment]
Reporting back with only top4 and top16 cuts I get the follwing results for the equi win rates:
![bye_point_scatter_eminence](https://github.com/Juizes-MTG-Portugal/Juizes-MTG-Portugal.github.io/assets/61587943/28ce7471-3505-4b2c-b93c-fdbca164479f)

4 points is always better than 5 points. But there are regions where 3 is better than 4 depending on the amount of players. Also sometimes TOs deviate from this given structure. Sometimes they play +-1 round, sometimes they make top16 instead of top4 cut or vice versa. Which is often due to venue and time contraints and can't be mitigated. But there could be some sort of catalouge (probably an interactive web app because listing all possible variations seems tedious) where given an abritrary structure you would get the amount of bye points that fits best.

Noteworthy is the region from 33-64 players, where for the most part even the 3/4 points bye has equi win rates of above 60%. The jump from 4 rounds+top4 to 5rounds+top16 doesn't seem to be be best choice regarding the fairness of byes at least. In future simulations I will take a look at alternative cuts and a top10 may provide a good alternative even in that region. Remains to be seen.

**Comment by fbatista** [comment]
Seems like the best solution is to "bite the 3-person-pods bullet".

Why?

Because in a highly competitive tournament setting, with highly competent players, the last round of swiss might only be played by players that still have a chance to make top cut.

In this scenario, the possibility of having at 3 byes, is extremely destructive to the tournament's competitive spirit and integrity.

The only solution in this case would be 3-person pods.

Another argument in favor of 3-person pods:

- It's something that can ALWAYS happen: A player might receive a penalty early / before game starts.
- A player might "no-show" for the match.

So despite the format being stable and balanced with 4 players in mind, 3-players are a reality that we can't ever escape from, we might as well embrace it and increase the overall quality of tournaments by eliminating the problematic byes.

I would finally just add that there should be a rule in the pairing algorithm that prevents players from playing in a 3-person pod more than once per event.

With this in place, i believe that 3-person pods are a better solution than byes.

**Comment by fbatista** [comment]
So the proposal will be:

---

If the tournament has 3 or more rounds in the "swiss portion" of the event, apply the following model:

Round 1, random pairing.
Round 2, snake pairing, the same as used in the Single Elimination portion.
Round 3+ "Swiss" pairing.

Top cut for Single Elimination.

For the "Swiss" pairing:

1. Pairing by Standing: [1,2,3,4] [5,6,7,8] [9,10,11,12] ... [Nth-2, Nth-1, Nth] (group of 3 at the end), 
  1.1. but try to avoid rematching, meaning there's a possibility of swapping players between groups, doing pair-up / pair down.
2. Then, ensure that in the last groups, the players who have already played in the group of 3 can be swapped with others who haven't played yet,
  2.1. but ensuring that there's no rematching in this situation. (in other words, repeat the process similar to the one used in point 1.1., but executed in reverse mode, starting from the last group).

Scoring: win = 5; loss = 0; draw = 1

Tiebreaker: 

- Opponent's Match Win %, but each player contributes at least (points for draw / points for win) = 0.2 (20%) to the OPP Match Win % of the opponents.
- In a group of 3, a 4th player is considered who will ALWAYS have 20% (points for draw / points for win) win% for the purposes of calculating the OPPONENT's Match Win %.

**Comment by fbatista** [comment]
In light of the recent developments, the proposal is to be changed to still keep the same pairing algorithgm changes as [described above](https://github.com/Juizes-MTG-Portugal/Juizes-MTG-Portugal.github.io/issues/2#issuecomment-1912351617) but introduce also the idea of using 3-player pods instead of byes.

The algorithm should also avoid putting a player in the 3-player pod more than once per tournament, and perform the necessary pair-up-downs.

**Comment by purplejudge** [comment]
> In light of the recent developments, the proposal is to be changed to still keep the same pairing algorithgm changes as https://github.com/Juizes-MTG-Portugal/Juizes-MTG-Portugal.github.io/issues/2#issuecomment-1912351617 but introduce also the idea of using 3-player pods instead of byes.

> The algorithm should also avoid putting a player in the 3-player pod more than once per tournament, and perform the necessary pair-up-downs.

Yup, sounds great. We're here to play Magic: the more we play, the better. If you are unhappy playing three-people pods, win a match and be done with it :)

**Comment by fbatista** [comment]
Since we've already merged the recommendation for not using byes, i'm going to close this and open a new issue with a proposal for the pairing algorithm changes that try to introduce more entropy into the winners pool and thus reduce the effectiveness of early intentional draws.
